x-pack/plugins/fleet/public/plugin.ts

```diff
@@ -239,36 +239,38 @@ export class FleetPlugin implements Plugin<FleetSetup, FleetStart, FleetSetupDep
   public start(core: CoreStart): FleetStart {
     let successPromise: ReturnType<FleetStart['isInitialized']>;
     const registerExtension = createExtensionRegistrationCallback(this.extensions);
+    const permissionsResponsePromise = core.http.get<CheckPermissionsResponse>(
```

Comment:
Could we not get an unhandled promise rejection here?

Reply:
Good catch, fixed this up and also took the opportunity to refactor `isInitialized` a bit to make it simpler.